### ISSUE TITLE
Create pass-through between views (i.e., points of intersection, connection, disagreement) between disciplines

### ISSUE THREAD

**Issue by tiagosousagarcia** [NewcastleRSE/beeing-human-web#24]
### Summary

_Create a method that allows people to jump through one view to another at a particular place (i.e., to see a different perspective on the subject). This would highlight point of contact and disagreement between the views._

### What needs to be done?

- [x] _Design the UI for the pass-through - a popup hover showing snippets of relevant text and including a button to take the reader there_
  - [x] Component that will indicate a destination for a portal (with a unique ID so that Portal origins won't need a lot data to get it right)
  - [x] Component that will create the link/button indicating a passthrough
  - [x] Component that will house the snippets -- maybe [this](https://tailwindui.com/components/application-ui/overlays/slide-overs#component-3e8348c3c183bd14fceb018d4cca1942)?
  - [x] Component for snippets
- [x] _Create standards to indicate this in the .md files_
- [x] _Documentation_
- [x] _Testing_

### Updates

20240501 - _Finished E2E and all other testing -- TODO: documentation_
20240429 - Nearly finished with E2E testing -- one test needs to be refined, another (very similar test) needs to be created; after that, TODO: documentation
20240424 - Finished unit and component testing, nearly done with E2E testing -- currently one E2E test failing that needs to be adjusted;
20240415 - _Started error testing (branch `testing-portals`)_
20240410 - _Previews for everything are now fetched through APIs; scroll to buzzword portal also works now, but a weird bug has been introduced that will scroll unnecessarily if the portal link is to the connections page itself_
20240409 - _Fetching previews of portals in buzzwords now works through an API; content of Portal in buzzwords is also fetched using an xml parser ([`cheerio`](https://cheerio.js.org/)). TODO: scroll to the buzzword portal only after the entire page has loaded_
20240408 - _Changed the buzzwords page into an API -- TODO: need to remove the currently used REGEX and use an XML parser instead to find the <Portal> elements; refactor the code in the PortalDestinationListCard to act differently with buzzwords_
20240326 - _Impossible to pre-load content from Portals in buzzwords as it stands: solution is to prerender all portals inside buzzwords and use that to generate the preview -- currently using the path /connections/buzzwords for that; buzzword portal links will need some sort of flag to change the expected link, but should be fine. Currently, the api-like buzzword page is getting the Tags -- need to convert it to html so that the Portal preview can fetch it;_
20240325 - _Trouble pre-loading content from Portals in buzzwords (though linking from buzzwords works fine)_
20240318 - _Fixed bug with connections img source; trying techniques to dynamically import buzzwords as modules, using mdsvex to convert_
20240314 - _After merging branch #60 `Portal` works fine inside `.md` files. Corrected an error caused by `mdsvex` replacing quotation marks in variable strings. TODO: correct `Portal` preview in the connections view (error in path for images in buzzwords) -- ignoring images could probably be enough for now_
20240312 - _Finished writing in-line documentation for components. It looks like we might need to change how content is loaded to be able to use the component in the `*.md` files. Will create a new issue and branch just to explore these options: #59 ._
20240228 - _Created some basic methods to test formation, tested automatic creation of links_
20240227 - _Created basic components to identify and display portals: `Portal` is used as both a portal point of origin (i.e., linking outwards) and point of destination (what other portals point to); `PortalPanel` is a side panel that shows `onclick` of a Portal (as long as it is origin or both) and giving a list of destinations; `PortalDestinationListCard` is the element that actually displays the snippets and gives the link. TODO: error catching to ensure `Portal` is always of the supported type; develop the server-side script that will collect all the information necessary to build the destination list; find a way to indicate Portals in the markdown and TEI files, and to transform them into Portal components._
